frontend/packages/dev-console/src/components/pipelines/pipeline-form/StartPipelineAdvancedSection.tsx

```diff
@@ -0,0 +1,93 @@
+import * as React from 'react';
+import { Formik } from 'formik';
+import { PlusCircleIcon } from '@patternfly/react-icons';
+import { Button } from '@patternfly/react-core';
+import { ExpandCollapse } from '@console/internal/components/utils';
+import { SecretModel } from '@console/internal/models';
+import { k8sCreate } from '@console/internal/module/k8s';
+import { SecretType } from '@console/internal/components/secrets/create-secret';
+import FormSection from '../../import/section/FormSection';
+import { associateServiceAccountToSecret } from '../../../utils/pipeline-utils';
+import SecretsList from './SecretsList';
+import SecretForm from './SecretForm';
+import './StartPipelineAdvancedSection.scss';
+
+type StartPipelineAdvancedSectionProps = {
+  namespace: string;
+};
+
+const initialValues = {
+  secretName: '',
+  type: SecretType.dockerconfigjson,
+  formData: {},
+};
+
+const StartPipelineAdvancedSection: React.FC<StartPipelineAdvancedSectionProps> = ({
+  namespace,
+}) => {
+  const [addSecret, setAddSecret] = React.useState(false);
+
+  const handleSubmit = (values, actions) => {
+    actions.setSubmitting(true);
+    const newSecret = {
+      apiVersion: SecretModel.apiVersion,
+      kind: SecretModel.kind,
+      metadata: {
+        name: values.secretName,
+        namespace,
+      },
+      type: values.type,
+      stringData: values.formData,
+    };
+
+    k8sCreate(SecretModel, newSecret)
+      .then((resp) => {
+        actions.setSubmitting(false);
+        setAddSecret(false);
+        associateServiceAccountToSecret(resp, namespace);
+      })
+      .catch((err) => {
+        actions.setSubmitting(false);
+        actions.setStatus({ submitError: err.message });
+      });
+  };
+
+  const handleReset = (values, actions) => {
+    actions.resetForm({ values: initialValues, status: {} });
+    setAddSecret(false);
+  };
+
+  const handleAddSecret = () => {
+    setAddSecret(true);
+  };
+
+  return (
+    <FormSection title="Advanced Options" fullWidth>
+      <ExpandCollapse
+        textExpanded="Hide Credential Options"
+        textCollapsed="Show Credential Options"
+      >
+        <div className="odc-start-pipeline-advanced-section">
+          <p>The following secrets are available for all pipelines in the namespace:</p>
+          <div className="odc-start-pipeline-advanced-section__secrets">
+            <SecretsList namespace={namespace} />
+            {!addSecret && (
+              <Button variant="link" onClick={handleAddSecret} icon={<PlusCircleIcon />}>
+                Add Secret
+              </Button>
+            )}
+            {addSecret && (
+              <div className="odc-start-pipeline-advanced-section__secret-form">
```

Comment:
```suggestion
            {!addSecret ? (
              <Button variant="link" onClick={handleAddSecret} icon={<PlusCircleIcon />}>
                Add Secret
              </Button>
            ) : (
              <div className="odc-start-pipeline-advanced-section__secret-form">
```
Probably worth inverting that check too, this is just an example.